.github/workflows/dotnet-upgrade.yml

```diff
@@ -0,0 +1,178 @@
+
+          should_update=0
+          [ $current_patch_version != $latest_patch_version ] && should_update=1
+
+          # check if git branch already exists for the upgrade
+          branch_already_exists=0
+          
+          if git ls-remote --heads --exit-code origin refs/heads/feature/dotnetsdk-upgrade/${latest_patch_version};
+          then 
+            branch_already_exists=1
+            should_update=0
+          fi
+          echo "DOTNET_LATEST_MAJOR_MINOR_PATCH_VERSION=${latest_patch_version}" >> $GITHUB_OUTPUT
+          echo "SHOULD_UPDATE=${should_update}" >> $GITHUB_OUTPUT
+          echo "BRANCH_EXISTS=${branch_already_exists}" >> $GITHUB_OUTPUT
+      - name: Create an error annotation if branch exists
+        if: ${{ steps.fetch_latest_version.outputs.BRANCH_EXISTS == 1 }}
+        run: echo "::error ::Branch feature/dotnetsdk-upgrade/${{ steps.fetch_latest_version.outputs.DOTNET_LATEST_MAJOR_MINOR_PATCH_VERSION }} already exists. Please take a look and delete that branch if you wish to recreate"
+      - name: Create a warning annotation if no need to update
+        if: ${{ steps.fetch_latest_version.outputs.SHOULD_UPDATE == 0 && steps.fetch_latest_version.outputs.BRANCH_EXISTS == 0 }}
+        run: echo "::warning ::Latest DotNet SDK patch is ${{ steps.fetch_latest_version.outputs.DOTNET_LATEST_MAJOR_MINOR_PATCH_VERSION }}, and we are on ${{ steps.fetch_latest_version.outputs.DOTNET_CURRENT_MAJOR_MINOR_PATCH_VERSION }}. No need to update"
+      - name: Update patch version
+        if: ${{ steps.fetch_latest_version.outputs.SHOULD_UPDATE == 1 && steps.fetch_latest_version.outputs.BRANCH_EXISTS == 0 }}
+        shell: bash
+        run: |
+          patch_version="${{ steps.fetch_latest_version.outputs.DOTNET_LATEST_MAJOR_MINOR_PATCH_VERSION }}"
+          current_version="${{ steps.fetch_current_version.outputs.DOTNET_CURRENT_MAJOR_MINOR_PATCH_VERSION }}"
+
+          # Update globals
+          cat ./src/global.json | jq --arg patch_version "$patch_version" '.sdk.version = $patch_version' | tee ./src/global.json
+
+          # Update devcontainer
+          cat ./.devcontainer/devcontainer.json | jq --arg patch_version "$patch_version" '.features."ghcr.io/devcontainers/features/dotnet".version = $patch_version' | tee ./.devcontainer/devcontainer.json
+
+          # Update dev.sh
+          sed -i "s/DOTNETSDK_VERSION=\"$current_version\"/DOTNETSDK_VERSION=\"$patch_version\"/g" ./src/dev.sh
+      - name: GIT commit and push all changed files
+        if: ${{ steps.fetch_latest_version.outputs.SHOULD_UPDATE == 1 && steps.fetch_latest_version.outputs.BRANCH_EXISTS == 0 }}
+        id: create_branch
+        run: |
+          branch_name="feature/dotnetsdk-upgrade/${{ steps.fetch_latest_version.outputs.DOTNET_LATEST_MAJOR_MINOR_PATCH_VERSION }}"
+          git config --global user.name "AutoUpdater"
+          git config --global user.email "username@users.noreply.github.com"
```

Comment:
We use this `github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>` in ARC, which makes the commit author look like:
![CleanShot 2023-08-25 at 12 06 50](https://github.com/actions/runner/assets/568794/018b1784-71dc-406d-9de7-2b3552c37cd5)